core/src/reactComponentHelpers.js

```diff
@@ -0,0 +1,61 @@
+// This is a temporary file. I made it so that I can update React Text Mask without breaking
+// other components. I'll have to merge it back into the original `componentHelpers.js` later.
+
+import adjustCaretPosition from './adjustCaretPosition.js'
+import conformToMask from './conformToMask.js'
+import {convertMaskToPlaceholder, isString} from './utilities.js'
+
+export function processComponentChanges({
+  userInput = '',
+  componentPlaceholder: placeholder = '',
+  previousConformedInput = '',
+  mask = '',
+  guide = '',
+  validator,
+  currentCaretPosition,
+  placeholderCharacter: placeholderChar
+}) {
+  const conformToMaskResults = conformToMask(
+    userInput,
+    mask,
+    {previousConformedInput, guide, placeholderChar, validator}
+  )
+  const {output: outputOfConformToMask} = conformToMaskResults
+  const adjustedCaretPosition = (currentCaretPosition !== undefined) ?
+    adjustCaretPosition({
+      previousConformedInput,
+      conformToMaskResults,
+      currentCaretPosition,
+      placeholderChar
+    }) :
+    undefined
+  const valueShouldBeEmpty = outputOfConformToMask === placeholder && adjustedCaretPosition === 0
+  const conformedInput = (valueShouldBeEmpty) ? '' : outputOfConformToMask
+
+  return {conformedInput, adjustedCaretPosition}
+}
+
+export function getComponentInitialState({mask, placeholderChar, inputValue}) {
+  if (inputValue !== undefined) {
+    checkInputValueType(inputValue)
+  }
```

Comment:
Also if we want to keep it we should remove in production, and only have this in DEV